### DISCUSSION TITLE
Template Repositories - Include labels, webhooks, etc.

### DISCUSSION THREAD

**Discussion by naomi-lgbt** [community/community#15512] [Other Feature Feedback, Questions, & Ideas]
The template repository is incredibly helpful, and I use it when creating almost every new project on GitHub.

One thing that's missing though: I still have to manually set up my labels and webhook. It would be nice if there was an option to carry over some settings, such as these - organisations have a default labels option, but on a personal account this isn't available.

**Comment by ClaudioFriederich** [comment]
In order to be actually useful, template repositories must carry over not only issue labels or webhooks, but **all** settings, including discussions and their icons, descriptions, categories, etc. and all settings on the Settings page.  Right now, there is no way to allow access to issues or discussions in a repository without granting access to its files.  All good and fine for free open source software, but for proprietary commercial software, you need a second repository, just to contain non-code artifacts like issues and discussions.  So you need a templating arrangement for these so that you can create such a companion repository each time you create a repository for a software artifact.  Since repository templates only carry over their files, you are forced to painstakingly recreate all the settings by hand for such non-code repositories.  Without templating capability for non-file settings, the whole issues, discussions, etc. system beyond file artifacts is like a screwdriver without a handle.
